rasa/nlu/test.py

```diff
@@ -398,6 +398,33 @@ def evaluate_response_selections(
     }
 
 
+def add_confused_intents_to_report(
+    report: Dict, target_intents: Iterable[Any], predicted_intents: Iterable[Any]
+) -> Dict:
+
+    from sklearn.metrics import confusion_matrix
+    from sklearn.utils.multiclass import unique_labels
+
+    cnf_matrix = confusion_matrix(target_intents, predicted_intents)
```

Comment:
We potentially end up calculating the confusion matrix  twice (also in line 504). Would be great if could to the confusion matrix calculating just once.